documentation/Using-the-SDK/Advanced/Creating-Your-Own-DataSource.md

```diff
@@ -0,0 +1,204 @@
+# Abstract
+
+This document outlines a scenario where you want to implement a Data Source
+beyond the FileDataSource and DirectoryDataSource provided by the SDK.
+
+# Motivation
+
+For users that have data stored somewhere other than Files or Directories, the
+ability to have their Data Source treated the same as a first class Data Source
+by the SDK is desireable. For example, let's suppose that you are writing many
+Custom Data Sources that get their data from a SQL database, or from a web
+service. Rather than having a file with a URI in it, it would be nice to say
+that your Custom Data Source uses a SqlServerDataSource, or a 
+WebServiceDataSource.
+
+# Requirements
+
+In order to have your new Data Source recognized by the SDK, you must do the
+following:
+
+- Create a new class inheriting from  _DataSource_ (or implementing _IDataSource_)
+- Create a new Attribute inheriting from _DataSourceAttribute_ specifying the
+  Type of the class you created
+- Optionally implement the _Accepts_ method on your new attribute.
+
+# Implementation
+
+This section is going to outline implementing a new Data Source Type to be
+consumed by Custom Data Sources. For the purposes of this Walkthrough, we will
+be implementing a Data Source backed by a WebService. This example is being
+implemented such that all three aspects are represented; there are multiple way
```

Comment:
this last sentence doesn't make sense to me. what is it saying?

Reply:
It uses the class, attribute, and overrides Accepts. I'll reword it.